pygsp/tests/test_graphs.py

```diff
@@ -319,5 +322,159 @@ def test_imgpatches(self):
     def test_grid2dimgpatches(self):
         graphs.Grid2dImgPatches(img=self._img, patch_shape=(3, 3))
 
-
 suite = unittest.TestLoader().loadTestsFromTestCase(TestCase)
+
+class TestCaseImportExport(unittest.TestCase):
+
+    def test_networkx_export_import(self):
+        #Export to networkx and reimport to PyGSP
+
+        #Exporting the Bunny graph
+        g = graphs.Bunny()
+        g_nx = g.to_networkx()
+        g2 = graphs.Graph.from_networkx(g_nx)
+        np.testing.assert_array_equal(g.W.todense(), g2.W.todense())
+
+    def test_networkx_import_export(self):
+        #Import from networkx then export to networkx again
+        g_nx = nx.gnm_random_graph(100, 50) #Generate a random graph
+        g = graphs.Graph.from_networkx(g_nx).to_networkx()
+
+        assert nx.is_isomorphic(g_nx, g)
+        np.testing.assert_array_equal(nx.adjacency_matrix(g_nx).todense(),
+                                      nx.adjacency_matrix(g).todense())
+
+    def test_graphtool_export_import(self):
+        #Export to graph tool and reimport to PyGSP directly
+        #The exported graph is a simple one without an associated Signal
+        g = graphs.Bunny()
+        g_gt = g.to_graphtool()
+        g2 = graphs.Graph.from_graphtool(g_gt)
+        np.testing.assert_array_equal(g.W.todense(), g2.W.todense())
+
+
+    def test_graphtool_multiedge_import(self):
+        #Manualy create a graph with multiple edges
+        g_gt = gt.Graph()
+        g_gt.add_vertex(10)
+        #connect edge (3,6) three times
+        for i in range(3):
+            g_gt.add_edge(g_gt.vertex(3), g_gt.vertex(6))
+        g = graphs.Graph.from_graphtool(g_gt)
+        assert g.W[3,6] == 3.0
+
+        #test custom aggregator function
+        g2 = graphs.Graph.from_graphtool(g_gt, aggr_fun=np.mean)
+        assert g2.W[3,6] == 1.0
+
+        eprop_double = g_gt.new_edge_property("double")
+
+        #Set the weight of 2 out of the 3 edges. The last one has a default weight of 0
+        e = g_gt.edge(3,6, all_edges=True)
+        eprop_double[e[0]] = 8.0
+        eprop_double[e[1]] = 1.0
+
+        g_gt.edge_properties["weight"] = eprop_double
+        g3 = graphs.Graph.from_graphtool(g_gt, aggr_fun=np.mean)
+        assert g3.W[3,6] == 3.0
+
+    def test_graphtool_import_export(self):
+        # Import to PyGSP and export again to graph tool directly
+        # create a random graphTool graph that does not contain multiple edges and no signal
+        g_gt = gt.Graph()
+        g_gt.add_vertex(100)
+
+        # insert single random links
+        eprop_double = g_gt.new_edge_property("double")
+        for s, t in set(zip(np.random.randint(0, 100, 100),
+                        np.random.randint(0, 100, 100))):
+            g_gt.add_edge(g_gt.vertex(s), g_gt.vertex(t))
```

Comment:
Thank you :)
I didn't know it was possible. I'll change the code.

Reply:
In fact I remember now why I didn't use it. graph_tool comes with a base and some [subpackages](https://graph-tool.skewed.de/static/doc/graph_tool.html#available-subpackages). In fact `graph_tool.generation` is a subpackage and need to be installed in addition. That's why I chose to generate the random graph randomly by myself instead of requiring more dependencies. Do you still want to use `graph_tool.generation.random_graph` ?

Reply:
I would use `graph_tool.generation`. Especially as this is for testing only, we don't require users to install anything. Did you have to install the subpackages separately? They were all there by default on my installation ([which is rather straightforward](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=python-graph-tool)).

Reply:
Yes I would have to install them separately. I followed this [tutorial](https://medium.com/@ronie/installing-graph-tool-for-python-3-on-anaconda-3f76d9004979). The installation process is really simple but doesn't contain the sub-packages.

Reply:
It does. Run `import graph_tool.generation`.